test/compress/loops.js

```diff
@@ -240,3 +240,81 @@ issue_1532: {
         }
     }
 }
+
+issue_186: {
+    beautify = {
+        screw_ie8: true,
+    }
+    input: {
+        var x = 3;
+        if (foo()) {
+            do {
+                do {
+                    alert(x);
+                } while (--x);
+            } while (x);
+        } else {
+            bar();
+        }
+    }
+    expect_exact: 'var x=3;if(foo())do do alert(x);while(--x)while(x)else bar();'
+}
+
+issue_186_ie8: {
+    beautify = {
+        screw_ie8: false,
+    }
+    input: {
+        var x = 3;
+        if (foo()) {
+            do {
+                do {
+                    alert(x);
+                } while (--x);
+            } while (x);
+        } else {
+            bar();
+        }
+    }
+    expect_exact: 'var x=3;if(foo())do do alert(x);while(--x)while(x)else bar();'
+}
+
+issue_186_beautify: {
+    beautify = {
+        beautify: true,
+        screw_ie8: true,
+    }
+    input: {
+        var x = 3;
+        if (foo()) {
+            do {
+                do {
+                    alert(x);
+                } while (--x);
+            } while (x);
+        } else {
+            bar();
+        }
+    }
+    expect_exact: 'var x = 3;\n\nif (foo()) do do alert(x); while (--x); while (x); else bar();'
+}
+
+issue_186_beautify_ie8: {
+    beautify = {
+        beautify: true,
+        screw_ie8: false,
+    }
+    input: {
+        var x = 3;
+        if (foo()) {
+            do {
+                do {
+                    alert(x);
+                } while (--x);
+            } while (x);
+        } else {
+            bar();
+        }
+    }
+    expect_exact: 'var x = 3;\n\nif (foo()) do do alert(x); while (--x) while (x) else bar();'
```

Comment:
@kzc does this look the same as https://github.com/mishoo/UglifyJS2/pull/937#issuecomment-229387433?

I reckon I'll take the risk this one time, but am also reading up on your suggested fix so if this does break I can quickly change it over.